editor/themes/editor_theme_manager.cpp

```diff
@@ -0,0 +1,2232 @@
+	config.border_width = CLAMP((int)EDITOR_GET("interface/theme/border_size"), 0, 2) * MAX(1, EDSCALE);
+	config.corner_radius = CLAMP((int)EDITOR_GET("interface/theme/corner_radius"), 0, 6);
+
+	config.draw_extra_borders = EDITOR_GET("interface/theme/draw_extra_borders");
+	config.relationship_line_opacity = EDITOR_GET("interface/theme/relationship_line_opacity");
+	config.thumb_size = EDITOR_GET("filesystem/file_dialog/thumbnail_size");
+	config.class_icon_size = 16 * EDSCALE;
+	config.increase_scrollbar_touch_area = EDITOR_GET("interface/touchscreen/increase_scrollbar_touch_area");
+	config.gizmo_handle_scale = EDITOR_GET("interface/touchscreen/scale_gizmo_handles");
+	config.color_picker_button_height = 28 * EDSCALE;
+
+	config.default_contrast = 0.3; // Make sure to keep this in sync with the editor settings definition.
+
+	// Handle main theme preset.
+	{
+		if (config.preset != "Custom") {
+			Color preset_accent_color;
+			Color preset_base_color;
+			float preset_contrast = 0;
+			bool preset_draw_extra_borders = false;
+
+			// Please use alphabetical order if you're adding a new theme here.
+			if (config.preset == "Breeze Dark") {
+				preset_accent_color = Color(0.26, 0.76, 1.00);
+				preset_base_color = Color(0.24, 0.26, 0.28);
+				preset_contrast = config.default_contrast;
+			} else if (config.preset == "Godot 2") {
+				preset_accent_color = Color(0.53, 0.67, 0.89);
+				preset_base_color = Color(0.24, 0.23, 0.27);
+				preset_contrast = config.default_contrast;
+			} else if (config.preset == "Gray") {
+				preset_accent_color = Color(0.44, 0.73, 0.98);
+				preset_base_color = Color(0.24, 0.24, 0.24);
+				preset_contrast = config.default_contrast;
+			} else if (config.preset == "Light") {
+				preset_accent_color = Color(0.18, 0.50, 1.00);
+				preset_base_color = Color(0.9, 0.9, 0.9);
+				// A negative contrast rate looks better for light themes, since it better follows the natural order of UI "elevation".
+				preset_contrast = -0.06;
+			} else if (config.preset == "Solarized (Dark)") {
+				preset_accent_color = Color(0.15, 0.55, 0.82);
+				preset_base_color = Color(0.04, 0.23, 0.27);
+				preset_contrast = config.default_contrast;
+			} else if (config.preset == "Solarized (Light)") {
+				preset_accent_color = Color(0.15, 0.55, 0.82);
+				preset_base_color = Color(0.89, 0.86, 0.79);
+				// A negative contrast rate looks better for light themes, since it better follows the natural order of UI "elevation".
+				preset_contrast = -0.06;
+			} else if (config.preset == "Black (OLED)") {
+				preset_accent_color = Color(0.45, 0.75, 1.0);
+				preset_base_color = Color(0, 0, 0);
+				// The contrast rate value is irrelevant on a fully black theme.
+				preset_contrast = 0.0;
+				preset_draw_extra_borders = true;
+			} else { // Default
+				preset_accent_color = Color(0.44, 0.73, 0.98);
+				preset_base_color = Color(0.21, 0.24, 0.29);
+				preset_contrast = config.default_contrast;
+			}
```

Comment:
Is there a reason for this not being in a switch statement?

Reply:
I don't know, I didn't change it. I guess it could be!

Reply:
> Is there a reason for this not being in a switch statement?

You can't use a String in a switch statement as far as I'm aware.

Reply:
Ah, you are correct, of course! Only integers and enums are supported by C++ switch statement. This could, of course, be an enum, but we only ever need it in one place, so it's probably not worth it.